alicloud/resource_alicloud_kvstore_backup_policy.go

```diff
@@ -0,0 +1,144 @@
+package alicloud
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/aliyun/alibaba-cloud-sdk-go/services/r-kvstore"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceAlicloudKVStoreBackupPolicy() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAlicloudKVStoreBackupPolicyCreate,
+		Read:   resourceAlicloudKVStoreBackupPolicyRead,
+		Update: resourceAlicloudKVStoreBackupPolicyUpdate,
+		Delete: resourceAlicloudKVStoreBackupPolicyDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+		Schema: map[string]*schema.Schema{
+			"instance_id": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"backup_time": &schema.Schema{
+				Type:         schema.TypeString,
+				ValidateFunc: validateAllowedStringValue(BACKUP_TIME),
+				Optional:     true,
+				Default:      "02:00Z-03:00Z",
+			},
+			"backup_period": &schema.Schema{
+				Type: schema.TypeSet,
+				Elem: &schema.Schema{Type: schema.TypeString},
+				// terraform does not support ValidateFunc of TypeList attr
+				// ValidateFunc: validateAllowedStringValue([]string{"Monday", "Tuesday", "Wednesday", "Thursday", "Friday", "Saturday", "Sunday"}),
+				Optional: true,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceAlicloudKVStoreBackupPolicyCreate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*AliyunClient)
+	conn := client.rkvconn
+
+	request := r_kvstore.CreateModifyBackupPolicyRequest()
+	request.InstanceId = d.Get("instance_id").(string)
+	request.PreferredBackupTime = d.Get("backup_time").(string)
+	periodList := expandStringList(d.Get("backup_period").(*schema.Set).List())
+	backupPeriod := fmt.Sprintf("%s", strings.Join(periodList[:], COMMA_SEPARATED))
+	request.PreferredBackupPeriod = backupPeriod
+
+	err := resource.Retry(5*time.Minute, func() *resource.RetryError {
+		if _, err := conn.ModifyBackupPolicy(request); err != nil {
+			return resource.NonRetryableError(fmt.Errorf("Create backup policy got an error: %#v", err))
+		}
+		return nil
+	})
+
+	if err != nil {
+		return err
+	}
+
+	// A security ip whitelist does not have a native IP.
+	d.SetId(request.InstanceId)
+
+	return resourceAlicloudKVStoreBackupPolicyRead(d, meta)
+}
+
+func resourceAlicloudKVStoreBackupPolicyRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*AliyunClient)
+	conn := client.rkvconn
+	instanceID := d.Id()
+
+	request := r_kvstore.CreateDescribeBackupPolicyRequest()
```

Comment:
I think describebackuppolicy action can be defined in the service_alicloud_rds.go and it can be multi-use by read method and testAccCheckKVStoreBackupPolicyExists, like DescribeRKVInstanceById.

Reply:
this method is actually defined in the official SDK. So not sure what you mean?

Reply:
Just like method `DescribeRKVInstanceById `